src/lib/components/NglMoleculeViewer.react.js

```diff
@@ -0,0 +1,286 @@
+            });
+
+            this.processDataFromBackend(data, stage, structuresList);
+        }
+    }
+
+    // styles the output of loadStructure/loadData
+    showStructure(stageObj, chain, color, xOffset, stage) {
+        if (chain !== 'ALL') {
+            const selection = new Selection(':' + chain);
+            const pa = stageObj.structure.getPrincipalAxes(selection);
+
+            stageObj.addRepresentation('cartoon', {
+                sele: ':' + chain,
+                color: color,
+            });
+            stageObj.setRotation(pa.getRotationQuaternion());
+
+            // translate by x angstrom along chosen axis
+            stageObj.setPosition([xOffset, 0, 0]);
+        } else {
+            stageObj.addRepresentation('cartoon');
+        }
+        stage.animationControls.moveComponent(stageObj, stageObj.getCenter());
+    }
+
+    // If user has selected structure already used before load it from the browser
+    loadStructure(stage, filename, chain, color, xOffset) {
+        const stageObj = stage.getComponentsByName(filename).list[0];
+        this.showStructure(stageObj, chain, color, xOffset, stage);
+    }
+
+    // If not load the structure from the backend
+    processDataFromBackend(data, stage, structuresList) {
+        const xval1 = 0;
+        const xval2 = 100;
+        const xval3 = 200;
+        const xval4 = 300;
+
+        const xOffsetArr = [xval1, xval2, xval3, xval4];
+        for (var i = 0; i < data.length; i++) {
+            const filename = data[i].filename;
+            if (structuresList.includes(filename)) {
+                this.loadStructure(
+                    stage,
+                    filename,
+                    data[i].chain,
+                    data[i].color,
+                    xOffsetArr[i]
+                );
+            } else {
+                this.loadData(data[i], stage, xOffsetArr[i]);
+            }
+        }
+        const center = stage.getCenter();
+        const newZoom = -500;
+        const duration = 1000;
+        stage.animationControls.zoomMove(center, newZoom, duration);
+    }
+
+    loadData(data, stage, xOffset) {
+        const stringBlob = new Blob([data.config.input], {
+            type: data.config.type,
+        });
+        stage
+            .loadFile(stringBlob, {ext: data.ext, defaultRepresentation: false})
+            .then(stageObj => {
+                stageObj.name = data.filename;
+                this.showStructure(
+                    stageObj,
+                    data.chain,
+                    data.color,
+                    xOffset,
+                    stage
+                );
+
+                this.setState(state => ({
+                    structuresList: state.structuresList.concat([
+                        data.filename,
+                    ]),
+                }));
+            });
+    }
+
+    render() {
+        const {id} = this.props;
+        return (
+            <div>
```

Comment:
Just wondering why there's a need for a wrapper `div`

Reply:
good catch removed the unnecessary wrapper